pkg/registry/core/service/strategy.go

```diff
@@ -104,6 +111,28 @@ func (svcStrategy) Export(ctx context.Context, obj runtime.Object, exact bool) e
 	return nil
 }
 
+// dropServiceDisabledFields drops fields that are not used if their associated feature gates
+// are not enabled
+func dropServiceDisabledFields(newSvc *api.Service, oldSvc *api.Service) {
+	// Drop IPFamily if DualStack is not enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && !serviceIPFamilyInUse(oldSvc) {
+		defaultServiceIPFamily := api.ServiceIPFamilyClusterDefault
+		newSvc.Spec.ServiceIPFamily = &defaultServiceIPFamily
+	}
+}
+
+// returns true if svc.Spec.ServiceIPFamily field is in use
+func serviceIPFamilyInUse(svc *api.Service) bool {
+	if svc == nil {
+		return false
+	}
+	// not used == set to default by defaulter (not to nil)
+	if svc.Spec.ServiceIPFamily != nil && *(svc.Spec.ServiceIPFamily) != api.ServiceIPFamilyClusterDefault {
```

Comment:
I don't think you cane make this assertion.  If it is not nil, it is in use.

Reply:
do we want to persist the value even when default is used?

Reply:
well, I am arguing there should not be a "Default" any more...

Reply:
i see your point, because we technically don't drop it, we set it to default.